### ISSUE TITLE
Troubles with running the simple example code on: https://diffeqflux.sciml.ai/dev/

### ISSUE THREAD

**Issue by ZhichaoPengMath** [SciML/DiffEqFlux.jl#616]
Hi,

I want to play with this package for my research. But I encountered some error with the very first example on the DiffEqFlux.jl webpage:

`using DiffEqSensitivity, OrdinaryDiffEq, Zygote

function fiip(du,u,p,t)
  du[1] = dx = p[1]*u[1] - p[2]*u[1]*u[2]
  du[2] = dy = -p[3]*u[2] + p[4]*u[1]*u[2]
end
p = [1.5,1.0,3.0,1.0]; u0 = [1.0;1.0]
prob = ODEProblem(fiip,u0,[0.0,10.0],p)
sol = solve(prob,Tsit5())
loss(u0,p) = sum(solve(prob,Tsit5(),u0=u0,p=p,saveat=0.1))

du01,dp1 = Zygote.gradient(loss,u0,p)`

I got the following error message:

`ERROR: LoadError: ArgumentError: tuple must be non-empty
Stacktrace:
 [1] first(::Tuple{}) at ./tuple.jl:95
 [2] _unapply(::Nothing, ::Tuple{}) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:163
 [3] _unapply(::Tuple{Nothing}, ::Tuple{}) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:167 (repeats 2 times)
 [4] _unapply(::Tuple{NTuple{4,Nothing},Tuple{Nothing}}, ::Tuple{Nothing,Array{Complex{Float64},1},Array{Complex{Float64},1},Nothing}) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:168
 [5] unapply(::Tuple{NTuple{4,Nothing},Tuple{Nothing}}, ::Tuple{Nothing,Array{Complex{Float64},1},Array{Complex{Float64},1},Nothing}) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:177
 [6] (::Zygote.var"#188#189"{Zygote.ZBack{DiffEqSensitivity.var"#adjoint_sensitivity_backpass#177"{Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},Tsit5,BacksolveAdjoint{0,true,Val{:central},ZygoteVJP,Bool},Array{Complex{Float64},1},Array{Float32,1},Tuple{},NamedTuple{(),Tuple{}},Colon}},Tuple{NTuple{4,Nothing},Tuple{Nothing}}})(::Array{Array{Complex{Float64},1},1}) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:195
 [7] (::Zygote.var"#1701#back#190"{Zygote.var"#188#189"{Zygote.ZBack{DiffEqSensitivity.var"#adjoint_sensitivity_backpass#177"{Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},Tsit5,BacksolveAdjoint{0,true,Val{:central},ZygoteVJP,Bool},Array{Complex{Float64},1},Array{Float32,1},Tuple{},NamedTuple{(),Tuple{}},Colon}},Tuple{NTuple{4,Nothing},Tuple{Nothing}}}})(::Array{Array{Complex{Float64},1},1}) at /Users/zhichaopeng/.julia/packages/ZygoteRules/OjfTt/src/adjoint.jl:59
 [8] #solve#57 at /Users/zhichaopeng/.julia/packages/DiffEqBase/QiFNl/src/solve.jl:70 [inlined]
 [9] (::typeof(∂(#solve#57)))(::Array{Array{Complex{Float64},1},1}) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/compiler/interface2.jl:0
 [10] (::Zygote.var"#188#189"{typeof(∂(#solve#57)),Tuple{NTuple{6,Nothing},Tuple{Nothing}}})(::Array{Array{Complex{Float64},1},1}) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:194
 [11] (::Zygote.var"#1701#back#190"{Zygote.var"#188#189"{typeof(∂(#solve#57)),Tuple{NTuple{6,Nothing},Tuple{Nothing}}}})(::Array{Array{Complex{Float64},1},1}) at /Users/zhichaopeng/.julia/packages/ZygoteRules/OjfTt/src/adjoint.jl:59
 [12] (::typeof(∂(solve##kw)))(::Array{Array{Complex{Float64},1},1}) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/compiler/interface2.jl:0
 [13] loss at /Users/zhichaopeng/Desktop/Github/LearnSciML.jl/test_diffeqflux.jl:14 [inlined]
 [14] (::typeof(∂(loss)))(::Float64) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/compiler/interface2.jl:0
 [15] #73 at /Users/zhichaopeng/.julia/packages/DiffEqFlux/Zt8Em/src/train.jl:86 [inlined]
 [16] #188 at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:194 [inlined]
 [17] #1701#back at /Users/zhichaopeng/.julia/packages/ZygoteRules/OjfTt/src/adjoint.jl:59 [inlined]
 [18] OptimizationFunction at /Users/zhichaopeng/.julia/packages/SciMLBase/DXiE6/src/problems/basic_problems.jl:107 [inlined]
 [19] #188 at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:194 [inlined]
 [20] #1701#back at /Users/zhichaopeng/.julia/packages/ZygoteRules/OjfTt/src/adjoint.jl:59 [inlined]
 [21] OptimizationFunction at /Users/zhichaopeng/.julia/packages/SciMLBase/DXiE6/src/problems/basic_problems.jl:107 [inlined]
 [22] #188 at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/lib/lib.jl:194 [inlined]
 [23] (::Zygote.var"#1701#back#190"{Zygote.var"#188#189"{typeof(∂(λ)),Tuple{Tuple{Nothing,Nothing},Int64}}})(::Float64) at /Users/zhichaopeng/.julia/packages/ZygoteRules/OjfTt/src/adjoint.jl:59
 [24] #8 at /Users/zhichaopeng/.julia/packages/GalacticOptim/paH6q/src/solve.jl:91 [inlined]
 [25] (::typeof(∂(λ)))(::Float64) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/compiler/interface2.jl:0
 [26] (::Zygote.var"#69#70"{Params,Zygote.Context,typeof(∂(λ))})(::Float64) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/compiler/interface.jl:255
 [27] gradient(::Function, ::Params) at /Users/zhichaopeng/.julia/packages/Zygote/zowrf/src/compiler/interface.jl:59
 [28] __solve(::OptimizationProblem{false,OptimizationFunction{false,GalacticOptim.AutoZygote,OptimizationFunction{true,GalacticOptim.AutoZygote,DiffEqFlux.var"#73#78"{typeof(loss)},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},GalacticOptim.var"#148#158"{GalacticOptim.var"#147#157"{OptimizationFunction{true,GalacticOptim.AutoZygote,DiffEqFlux.var"#73#78"{typeof(loss)},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Nothing}},GalacticOptim.var"#151#161"{GalacticOptim.var"#147#157"{OptimizationFunction{true,GalacticOptim.AutoZygote,DiffEqFlux.var"#73#78"{typeof(loss)},Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Nothing}},GalacticOptim.var"#156#166",Nothing,Nothing,Nothing},Array{Float32,1},SciMLBase.NullParameters,Nothing,Nothing,Nothing,Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}}, ::ADAM, ::Base.Iterators.Cycle{Tuple{GalacticOptim.NullData}}; maxiters::Int64, cb::Function, progress::Bool, save_best::Bool, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/zhichaopeng/.julia/packages/GalacticOptim/paH6q/src/solve.jl:90
 [29] #solve#468 at /Users/zhichaopeng/.julia/packages/SciMLBase/DXiE6/src/solve.jl:3 [inlined]
 [30] sciml_train(::typeof(loss), ::Array{Float32,1}, ::ADAM, ::Nothing; lower_bounds::Nothing, upper_bounds::Nothing, maxiters::Int64, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/zhichaopeng/.julia/packages/DiffEqFlux/Zt8Em/src/train.jl:91
 [31] top-level scope at /Users/zhichaopeng/Desktop/Github/LearnSciML.jl/test_diffeqflux.jl:18
 [32] include(::String) at ./client.jl:457
 [33] top-level scope at none:1
in expression starting at /Users/zhichaopeng/Desktop/Github/LearnSciML.jl/test_diffeqflux.jl:18`

I have Julia "1.5.4", Zygote 0.6.12, DiffEqSensitivity v6.49.1, OrdinaryDiffEq v5.55.1.  Zygote.gradient works fine, if I change the loss function as sum(u0)+sum(p).

If anyone would like to give me any hints, I will be more than grateful.  Thanks!

Best,
Zhichao

**Comment by ChrisRackauckas** [comment]
What about with Julia v1.6?

**Comment by ZhichaoPengMath** [comment]
Hi Chris,

Thanks for your comment! Last night, I cleaned my ~/.julia and reinstalled Julia with v1.6, DifferentialEquations.jl, DiffEqFlux.jl, Flux.jl and some other packages. The error was fixed. 

It might be some compatibility issues with Julia v1.5 or some packages I used. Let me know if you want me to check whether I can reproduce the error with lower versions of Julia. It might just be my packages instead of Julia itself.

I will close this issue for now.

Best,
Zhichao
